### PR TITLE
Added AWS Credentials Support for Scanning Private Registry

### PR DESCRIPTION
**Pull request by VF-mbrauer**
Added support for scanning the **private registry** of AWS-ECR.
Feature still in Beta-mode but usable when prerequisites are met.

**Prerequisite:**
K8S-Kiam or any other method IRSA (Future) to be used to allow Assuming Instance Roles.

**Configurations:**
Annotate the Starboard Namespace to be allowed to use KIAM:
`Annotations:  iam.amazonaws.com/permitted: .*`

Add the Role you want to Assume which has the proper right to create ECR-Credentials
```
podAnnotations: 
    # iam.amazonaws.com/role: <yourRolewithPermissions>
```

Configuration in HELM value file to enable the feature
`trivy.useEcrRoleCreds: false`

### CODE REVIEW

File: pkg/plugin/trivy/plugin.go [hunk 1]
```diff
@@ -1301,3 +1350,33 @@ func constructEnvVarSourceFromConfigMap(envName, trivyConfigName, trivyConfikey
...
+	result, err := svc.GetAuthorizationToken(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ecr.ErrCodeServerException:
+				fmt.Println(ecr.ErrCodeServerException, aerr.Error())
+			case ecr.ErrCodeInvalidParameterException:
+				fmt.Println(ecr.ErrCodeInvalidParameterException, aerr.Error())
+			default:
+				fmt.Println(aerr.Error())
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1367]: same here use logger

File: pkg/plugin/trivy/plugin.go [hunk 2]
```diff
@@ -1301,3 +1350,33 @@ func constructEnvVarSourceFromConfigMap(envName, trivyConfigName, trivyConfikey
...
+	input := &ecr.GetAuthorizationTokenInput{}
+
+	result, err := svc.GetAuthorizationToken(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ecr.ErrCodeServerException:
+				fmt.Println(ecr.ErrCodeServerException, aerr.Error())
+			case ecr.ErrCodeInvalidParameterException:
+				fmt.Println(ecr.ErrCodeInvalidParameterException, aerr.Error())
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1365]: same here use logger

File: pkg/plugin/trivy/plugin.go [hunk 3]
```diff
@@ -1301,3 +1350,33 @@ func constructEnvVarSourceFromConfigMap(envName, trivyConfigName, trivyConfikey
...
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1370]: ^

File: pkg/plugin/trivy/plugin.go [hunk 4]
```diff
@@ -1301,3 +1350,33 @@ func constructEnvVarSourceFromConfigMap(envName, trivyConfigName, trivyConfikey
...
+		}
+		return nil
+	}
+
+	var credentials = *result.AuthorizationData[0].AuthorizationToken
+
+	sDec, _ := base64.StdEncoding.DecodeString(credentials)
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1377]: `error` should not be ignored

**Review by chen-keinan — CHANGES_REQUESTED** [review]

File: pkg/plugin/trivy/plugin.go [hunk 5]
```diff
@@ -547,31 +568,45 @@ func (p *plugin) getPodSpecForStandaloneMode(ctx starboard.PluginContext, config
 			})
 		}
 
-		if _, ok := credentials[c.Name]; ok && secret != nil {
-			registryUsernameKey := fmt.Sprintf("%s.username", c.Name)
-			registryPasswordKey := fmt.Sprintf("%s.password", c.Name)
+		if config.UseECRCredentials() {
+			var aws_creds = GetAuthorizationToken(c.Image)
+			var creds (ecr_credentials) = ecr_credentials{aws_creds[0][1], aws_creds[0][2]}
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:573]: check matrix size both row and column to avoid `index out of bound`

File: pkg/plugin/trivy/plugin.go [hunk 6]
```diff
@@ -795,31 +830,45 @@ func (p *plugin) getPodSpecForClientServerMode(ctx starboard.PluginContext, conf
 			},
 		}
 
-		if _, ok := credentials[container.Name]; ok && secret != nil {
-			registryUsernameKey := fmt.Sprintf("%s.username", container.Name)
-			registryPasswordKey := fmt.Sprintf("%s.password", container.Name)
+		if config.UseECRCredentials() {
+			var aws_creds = GetAuthorizationToken(container.Image)
+			var creds (ecr_credentials) = ecr_credentials{aws_creds[0][1], aws_creds[0][2]}
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:835]: check matrix size both row and column to avoid `index out of range`

**Comment by VF-mbrauer** [comment]
Due to the new Release 0.15.0 the PR needed to be rebased. For this, a new PR has been created #1103
This PR can be closed.

**Pull request closed without merging**